app/V1Module/presenters/AssignmentSolutionsPresenter.php

```diff
@@ -520,6 +521,40 @@ public function actionDownloadSolutionArchive(string $id)
         $this->sendResponse(new StorageFileResponse($zipFile, "solution-{$id}.zip", "application/zip"));
     }
 
+    public function checkFiles(string $id)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id);
+        if (!$this->assignmentSolutionAcl->canViewDetail($solution)) {
+            throw new ForbiddenRequestException("You cannot access the solution files metadata");
+        }
+    }
+
+    /**
+     * Get the list of submitted files of the solution.
+     * @GET
+     * @param string $id of assignment solution
+     * @throws ForbiddenRequestException
+     * @throws NotFoundException
+     */
+    public function actionFiles(string $id)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id)->getSolution();
+        $entryPoint = $solution->getSolutionParams()->getVariable(ExecutionBox::$ENTRY_POINT_KEY);
+        $entryPoint = $entryPoint ? $entryPoint->getValue() : null;
+
+        $files = $solution->getFiles()->toArray();
+        foreach ($files as &$file) {
+            $file->prepareExtendedSerializationData($this->fileStorage);
+            $file = $file->jsonSerialize();
+            if ($file["name"] === $entryPoint) {
+                $file["isEntryPoint"] = true;
```

Comment:
This arbitrary change of the response JSON structure are probably hard to document, might be better to somehow avoid them as much as possible... Just a thought